packages/SwingSet/docs/vat-upgrade.md

```diff
@@ -128,23 +135,26 @@ initializeBar(barData);
 
 Once the governance object is holding the v2 bundlecap, it triggers the upgrade with `E(adminNode).upgrade(newBundlecap, options)`. This schedules the upgrade sequence (described above), and returns a Promise that resolves when the upgrade is complete. If the upgrade fails, the Promise is rejected and the old vat is reinstalled. An upgrade might fail because the new source bundle has a syntax error (preventing evaluation), or because the upgrade phase throws an exception or returns a Promise that rejects. It will also fail if the ugprade phase does not fulfill all of its obligations, such as leaving a durable Kind unattached.
 
 An important property of the `options` bag is `vatParameters`. This value is passed to the upgrade phase (as the usual second argument to `buildRootObject()`) and can be used to communicate with the upgrade-time code before any external messages have a chance to be delivered. In the Zoe ZCF vat, this is how new contract code will be delivered, so it can be executed (and can assume responsibility for v1 obligations) to completion by the time the upgrade phase finishes.
 
 ## From Inside: V2 Executes the Upgrade
 
-The v2 code wakes up inside the upgrade phase. The exact function signature is TBD, but for now let's assume that the v2 code is expected to provide the same `buildRootObject()` export as v1. In this case, the upgrade phase begins with `buildRootObject(vatPowers, vatParameters, baggage)` being called, where `vatParameters` will come from the call to `upgrade`. This `buildRootObject()` is expected to return an object, or a Promise that resolves to an object, and that object will assume the identity of the root object.
+The v2 code wakes up inside the upgrade phase when `buildRootObject(vatPowers, vatParameters, baggage)` is called, where `vatParameters` will come from the call to `upgrade`. This `buildRootObject()` is expected to return an object, or a Promise that resolves to an object, and that object will assume the identity of the root object.
 
-Before `buildRootObject()` returns (and/or its return Promise is resolved), the v2 code is obligated to call `makeDurableKind()` for every Kind that was created by the v1 code. It needs the stored handles to identify which Kind is being replaced:
+Before `buildRootObject()` returns (and/or its return Promise is resolved), the v2 code is obligated to redefine every Kind that was created by the v1 code.
 
 ```js
-const fooHandle = baggage.get('foo handle');
-const makeFoo = makeDurableKind(fooHandle, newFooBehavior);
+import { M } from '@agoric/store';
+// fooMethodGuards and fooMethods must match or be a superset of their v1 analogs.
```

Comment:
Is there anything that enforces that `fooMethodGuards` and `fooMethods` must match or be a superset of their v1 analogs

Reply:
Yes; I think it's in liveslots (and now mentioned in this file).